packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js

```diff
@@ -10,95 +10,671 @@
+      fireEvent.mouseMove(button);
+      expect(onOpenChange).not.toHaveBeenCalled();
+      expect(() => getByRole('tooltip')).toThrow();
+      // run half way through the timers and see if it's appeared
+      act(() => {jest.advanceTimersByTime(TOOLTIP_DELAY / 2);});
+      expect(onOpenChange).not.toHaveBeenCalled();
+      expect(() => getByRole('tooltip')).toThrow();
+      // finish the full amount of time, now it should be visible
+      act(() => {jest.advanceTimersByTime(TOOLTIP_DELAY / 2);});
+      expect(onOpenChange).toHaveBeenCalledWith(true);
+      let tooltip = getByRole('tooltip');
+      expect(tooltip).toBeVisible();
+      fireEvent.mouseLeave(button);
+      expect(onOpenChange).toHaveBeenCalledWith(false);
+      act(() => {
+        jest.advanceTimersByTime(CLOSE_TIME);
+      });
+      expect(tooltip).not.toBeInTheDocument();
+    });
+
+    it('never opens if blurred before it opens',  () => {
+      let {getByRole, getByLabelText} = render(
+        <Provider theme={theme}>
+          <TooltipTrigger delay onOpenChange={onOpenChange}>
+            <ActionButton aria-label="trigger" />
+            <Tooltip>Helpful information.</Tooltip>
+          </TooltipTrigger>
+        </Provider>
+      );
+
+      let button = getByLabelText('trigger');
+      act(() => {
+        button.focus();
+      });
+      expect(onOpenChange).not.toHaveBeenCalled();
+      expect(() => getByRole('tooltip')).toThrow();
+      // run half way through the timers and see if it's appeared
+      act(() => {jest.advanceTimersByTime(TOOLTIP_DELAY / 2);});
+      expect(onOpenChange).not.toHaveBeenCalled();
+      expect(() => getByRole('tooltip')).toThrow();
+      act(() => {
+        button.blur();
+      });
+      expect(onOpenChange).not.toHaveBeenCalled();
+      act(() => {
+        jest.runAllTimers();
+      });
+      expect(() => getByRole('tooltip')).toThrow();
+    });
 
-    it('pressing escape if the trigger is focused should close the tooltip', async function () {
-      let {getByText} = render(
+    it('opens for whichever occurs first',  () => {
+      let {getByRole, getByLabelText} = render(
         <Provider theme={theme}>
-          <TooltipTrigger type="hover">
-            <ActionButton>Trigger</ActionButton>
-            <Tooltip>content</Tooltip>
+          <TooltipTrigger delay>
+            <ActionButton aria-label="trigger" />
+            <Tooltip>Helpful information.</Tooltip>
           </TooltipTrigger>
         </Provider>
       );
 
-      let button = getByText('Trigger');
-      fireEvent.mouseOver(button);
+      let button = getByLabelText('trigger');
+      act(() => {
+        button.focus();
+      });
+      expect(() => getByRole('tooltip')).toThrow();
+      // run half way through the timers and see if it's appeared
+      act(() => {jest.advanceTimersByTime(TOOLTIP_DELAY / 2);});
+      expect(() => getByRole('tooltip')).toThrow();
+      // halfway through, now add a hover
```

Comment:
this test verifies that the first event that should open it will cause it to open TOOLTIP_DELAY after it occurred and that it won't be reset by a second event that should cause it to open
t = 0, focus (start timer)
t = TOOLTIP_DELAY / 2, mouseover (should not start a new timer)
t = TOOLTIP_DELAY, tooltip should show now
t= TOOLTIP_DELAY + TOOLTIP_DELAY / 2, tooltip should still be showing